Dockerfile

```diff
@@ -2,64 +2,49 @@
 # Build Image
 #
 
-FROM ubuntu:18.04 as builder
+FROM alpine:3.8 as builder
 
 # Add the git repo
 ADD . /git/mapcrafter
 
 # Dependencies needed for building Mapcrafter
-RUN apt-get update && \
-    apt-get -y install \
+# (not sure how many of these are actually needed)
+RUN apk add \
         cmake \
-        build-essential \
+        gcc \
+        make \
+        g++ \
+        zlib-dev \
         libpng-dev \
-        libjpeg-dev \
-        libboost-iostreams-dev \
-        libboost-system-dev \
-        libboost-filesystem-dev \
-        libboost-program-options-dev \
-        libboost-test-dev \
-        wget \
-        python \
-        imagemagick
-
-
-# Textures
-# With thanks to https://mcversions.net/ for the jar link
-RUN cd /git/mapcrafter && \
-    wget https://launcher.mojang.com/v1/objects/8de235e5ec3a7fce168056ea395d21cbdec18d7c/client.jar && \
-    ./src/tools/mapcrafter_textures.py client.jar src/data/textures/
+        libjpeg-turbo-dev \
+        boost-dev
 
 # Build mapcrafter from source
 RUN cd /git/mapcrafter && \
     mkdir build && cd build && \
     cmake .. && \
     make && \
     mkdir /tmp/mapcrafter && \
-    make DESTDIR=/tmp/mapcrafter install && \
-    ldconfig
+    make DESTDIR=/tmp/mapcrafter install
 
 
 #
 # Final Image
 #
 
-FROM ubuntu:18.04
+FROM alpine:3.8
 
 # Mapcrafter, built in previous stage
 COPY --from=builder /tmp/mapcrafter/ /
 
 # Depedencies needed for running Mapcrafter
-RUN apt-get update && \
-    apt-get -y install \
-        libpng16-16 \
-        libjpeg-turbo-progs \
-        libboost-iostreams1.65.1 \
-        libboost-system1.65.1 \
-        libboost-filesystem1.65.1 \
-        libboost-program-options1.65.1 && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
-    ldconfig
+RUN apk add \
+        libpng \
+        libjpeg-turbo \
+        boost \
+        boost-iostreams \
+        boost-system \
+        boost-filesystem \
+        boost-program_options
 
 ENTRYPOINT ["mapcrafter"]
```

Comment:
One more thing came to my mind: We have to make sure Mapcrafter runs as the same user inside of the docker container. Otherwise files are created as root and not writable by the user later on. I have fixed it already with Blockcrafter: https://github.com/mapcrafter/blockcrafter/blob/master/entrypoint.sh

Could you integrate this as well?

Reply:
Ah, yes. I have something similar to this in my own docker images.

I forgot about this, as most of my images run on Docker-for-Mac (which handles the file ownership automagically).

Added in c9fd3ad

Only brings the image up to 72.9MB

(Also fixes an issue I just noticed where it wasn't picking up a `ctrl-c`)